src/OrchardCore.Modules/OrchardCore.Users/Handlers/ScriptExternalLoginEventHandler.cs

```diff
@@ -54,6 +61,32 @@ public async Task UpdateRoles(UpdateRolesContext context)
                 dynamic evaluationResult = _scriptingManager.Evaluate(script, null, null, null);
                 context.RolesToAdd.AddRange((evaluationResult.rolesToAdd as object[]).Select(i => i.ToString()));
                 context.RolesToRemove.AddRange((evaluationResult.rolesToRemove as object[]).Select(i => i.ToString()));
+
+                var claimToAdDict = (evaluationResult.claimsToAdd as IDictionary<string, object>)?.ToDictionary(x => x.Key, v => v.Value.ToString());
```

Comment:
I don't think you need to clone the dictionary here (and below). Just use it directly.

Reply:
Yes, but now I want to change it to a collection of objects instead of a dictionary